crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+    let db = &mut RootDatabase::builder().build()?;
+
+    init_dev_corelib(db, PathBuf::from(corelib));
+
+    let main_crate_ids = setup_project(db, Path::new(&script_path))?;
+    if DiagnosticsReporter::stderr().check(db) {
+        anyhow::bail!("failed to compile: {}", script_path);
+    }
+
+    let sierra_program = db
+        .get_sierra_program(main_crate_ids.clone())
+        .to_option()
+        .with_context(|| "Compilation failed without any diagnostics.")?;
+    let replacer = DebugReplacer { db };
+
+    let runner = SierraCasmRunner::new(
+        replacer.apply(&sierra_program),
+        Some(MetadataComputationConfig::default()),
+        OrderedHashMap::default(),
+    )
+    .with_context(|| "Failed setting up runner.")?;
+
+    let func = runner.find_function("::main")?;
+    let (entry_code, builtins) = runner.create_entry_code(func, &Vec::new(), usize::MAX)?;
+    let footer = runner.create_code_footer();
+    let instructions = chain!(
+        entry_code.iter(),
+        runner.get_casm_program().instructions.iter(),
+        footer.iter()
+    );
+
+    // import from cairo-lang-runner
+    let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
+
+    // hint processor
+    let block_context = build_block_context(CheatnetBlockInfo::default());
+    let account_context = build_transaction_context();
+    let mut context = EntryPointExecutionContext::new(
+        &block_context.clone(),
+        &account_context,
+        ExecutionMode::Execute,
+    );
+
+    let mut blockifier_state =
+        CachedState::new(DictStateReader::default(), GlobalContractCache::default());
+    let mut execution_resources = ExecutionResources::default();
+
+    let syscall_handler = SyscallHintProcessor::new(
+        &mut blockifier_state,
+        &mut execution_resources,
+        &mut context,
+        // This segment is created by SierraCasmRunner
+        Relocatable {
+            segment_index: 10,
+            offset: 0,
+        },
+        CallEntryPoint::default(),
+        &string_to_hint,
+        ReadOnlySegments::default(),
+    );
+
+    let mut cairo_hint_processor = CairoHintProcessor {
+        blockifier_syscall_handler: syscall_handler,
+        hints: &string_to_hint,
+        provider,
+        runtime,
+    };
+
+    match runner.run_function(
+        runner.find_function("::main")?,
+        &mut cairo_hint_processor,
+        hints_dict,
+        instructions,
+        builtins,
+    ) {
+        Ok(result) => match result.value {
+            RunResultValue::Success(data) => Ok(ScriptResponse {
+                status: "success".to_string(),
+                msg: build_readable_text(&data),
+            }),
+            RunResultValue::Panic(panic_data) => Ok(ScriptResponse {
+                status: "script panicked".to_string(),
+                msg: build_readable_text(&panic_data),
+            }),
+        },
+        Err(err) => Err(err.into()),
+    }
+}
```

Comment:
This could be multiple smaller functions.

Reply:
extracted some logic to `compile_script` & we use scarb build now for compilation